var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -0,0 +1,95 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Lbann(CMakePackage):
+    """LBANN: Livermore Big Artificial Neural Network Toolkit.  A distributed memory, HPC-optimized, model and data parallel training toolkit for deep neural networks."""
+
+    homepage = "http://software.llnl.gov/lbann/"
+    url      = "https://github.com/LLNL/lbann/archive/v0.91.tar.gz"
+
+    version('develop', git='https://github.com/LLNL/lbann.git', branch="develop")
+    version('0.91', '83b0ec9cd0b7625d41dfb06d2abd4134')
+
+    variant('debug', default=False,
+            description='Builds a debug version of the libraries')
+    variant('gpu', default=False,
+            description='Builds with support for GPUs via CUDA and cuDNN')
+    variant('opencv', default=True,
+            description='Builds with support for image processing routines with OpenCV')
+
+    variant('shared', default=True,
+            description='Enables the build of shared libraries')
+    variant('int64', default=False,
+            description='Elemental: use 64bit integers')
+    variant('int64_blas', default=False,
+            description='Elemental: use 64bit integers for BLAS')
+
+    depends_on('cmake', type='build')
+
+    # Currently required to allow lbann to specify a specific BLAS library
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('scalapack')
+
+    depends_on('elemental +openmp_blas +scalapack')
+    depends_on('elemental +openmp_blas +scalapack +int64', when='+int64')
```

Comment:
@tgamblin This gets into the variant forwarding discussion.

If we let this case slide, then why not add `+pic` to every single package in Spack that depends on `zlib`? Obviously you wouldn't want that, and you can just do `hdf5 ^zlib+pic` if you needed to.

Another reason to avoid this is that variant forwarding is currently broken (_cough cough_).

Maybe it would be better to allow variants to appear anywhere on the command line, so that `foo +mpi` means activate the `mpi` variant for every package in the spec that has one.